sources/cassandra/README.md

```diff
@@ -0,0 +1,228 @@
+# Cassandra to Cloud Spanner Migration Components
+
+⚠️ **WARNING: This is a standalone component and is not integrated with the spanner-migration-tool. It must be used separately by cloning this repository.**
+
+## Overview
+
+This repository contains components for doing live migrations from Cassandra to Google Cloud Spanner.
+## Key Components
+
+### 1. [DataStax ZDM Proxy](https://github.com/datastax/zdm-proxy)
+- Acts as an intermediary between your application and databases
+- Handles dual writes to both Cassandra and Spanner
+- Ensures data consistency during migration
+
+### 2. Cassandra-Spanner Proxy
+- Runs as a sidecar to ZDM proxy
+- Translates CQL (Cassandra Query Language) to Spanner API calls
+- Enables Cassandra-compatible applications to interact with Spanner
+
+## Architecture
+
+![Cassandra to Spanner Migration Architecture](https://services.google.com/fh/files/misc/live-cassandra.png)
+
+
+The migration setup consists of:
+- **Origin**: Source Cassandra database
+- **Target**: Destination Cloud Spanner database
+- **ZDM Proxy**: Manages dual writes and read routing
+- **Cassandra-Spanner Proxy**: Translates CQL to Spanner API calls
+- **Client Application**: Your application that interacts with the databases
+
+## Migration Process
+
+1. **Infrastructure Setup**
+   - Configure Spanner database
+   - Deploy ZDM proxy with sidecar
+   - Establish network connectivity
+
+2. **Application Configuration**
+   - Point application to ZDM proxy
+   - Verify dual-write functionality on Cassandra and Spanner
+
+3. **Data Migration**
+   - Execute [bulk data migration](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/sourcedb-to-spanner/README_Sourcedb_to_Spanner.md#cassandra-to-spanner-bulk-migration)
+   - Monitor progress and performance
+
+4. **Validation**
+   - Perform data validation over row counts and row data
+   - Reconcile any inconsistencies
+   - Refer to the [validation scripts](./validations/) for validating data consistency between source and target clusters.
+
+5. **Cutover**
+   - Switch application to direct Spanner connection
+   - Decommission proxies
+
+## Prerequisites
+
+- Docker (for development/testing)
+- Terraform (for production deployment)
+- Google Cloud Platform project with Spanner instance
+- Service account with appropriate permissions
+- Network connectivity between components
+
+## Getting Started
+
+### Single Instance Testing Setup
+
+This guide helps you set up a single instance of the proxy for development or testing purposes using Docker.
+
+#### Prerequisites
+
+Before you begin, ensure you have:
+- Docker installed on your system
+- Network connectivity between:
```

Comment:
If there is a way to test this connectivity, it would help to link it here.

Reply:
No links that can help test, we can test these easily via cqlsh though. Ill add a line fo rit